pkg/receive/handler.go

```diff
@@ -295,12 +303,16 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	err = h.handleRequest(ctx, rep, tenant, &wreq)
-	switch err {
+	if err != nil {
+		level.Debug(h.logger).Log("msg", "failed to handle request", "err", err)
+	}
+
+	switch rootCause(err) {
 	case nil:
 		return
-	case tsdb.ErrNotReady:
```

Comment:
Why this? (:

Reply:
The new error encapsulates both of the cases, tsdb not ready and service unavailable. IMHO It's better not to expose a sentinel error that belongs to a dependency.

Reply:
The new error encapsulates both of the cases, tsdb not ready and service unavailable. 

Is this what we want? :thinking: I think then `errUnavailable` might more precise :thinking: 

>  IMHO It's better not to expose a sentinel error that belongs to a dependency.

No strong feelings, you might be right